tests/MongoDB.EntityFrameworkCore.UnitTests/ValueGeneration/OwnedEntityIndexValueGeneratorTests.cs

```diff
@@ -0,0 +1,34 @@
+﻿/* Copyright 2023-present MongoDB Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+using MongoDB.EntityFrameworkCore.ValueGeneration;
+
+namespace MongoDB.EntityFrameworkCore.UnitTests.ValueGeneration;
+
+public class OwnedEntityIndexValueGeneratorTests
+{
+    [Fact]
+    public void Next_generates_non_temporary_id()
```

Comment:
Should we add tests to ensure no duplicates generated for concurrent execution?

Reply:
We could but we don't currently have any of that level of test coverage and this interlock pattern is what they use throughout EF Core.